app/src/org/commcare/tasks/ProcessAndSendTask.java

```diff
@@ -68,12 +76,13 @@ public ProcessAndSendTask(Context c, String url, boolean inSyncMode) {
         this.c = c;
         this.url = url;
         this.processor = new FormRecordProcessor(c);
+        this.formSubmissionListeners = new ArrayList<>();
         if (inSyncMode) {
             this.sendTaskId = SEND_PHASE_ID;
             this.taskId = PROCESSING_PHASE_ID;
         } else {
-            this.sendTaskId = -1;
-            this.taskId = -1;
+            this.sendTaskId = SEND_PHASE_ID_NO_DIALOG;

```

Comment:
Already off-lined with @amstone326 about this and agreed out of the purview of this PR but seems like splitting the process and send portions of this task would be a good refactor. The usage of ID to denote state here gets a bit confusing/messy IMO